src/sentry/static/sentry/app/components/forms/apiForm.tsx

```diff
@@ -40,14 +41,32 @@ export default class ApiForm extends Form<Props> {
     this.api.clear();
   }
 
+  getEnabledData() {
+    // Return a hash of data from non-disabled fields.
+    const data = {...this.state.data}; // copy to avoid mutating original
```

Comment:
Alternative approach: start with an empty object and only add valid ones instead of deletion.

Stating this as for some reason my brain assumes `delete` operation is costly or _bad_. Old habits maybe?

Btw. when `this.props.children` is not an array, now we return the full `this.state.data` object. Is that expected?

Reply:
Some digging turns up:

> In the below example, `delete o.x` does a lot more harm than good behind the scenes, as it changes `o`’s hidden class and makes it a generic slow object.

https://www.smashingmagazine.com/2012/11/writing-fast-memory-efficient-javascript/

2012, so ¯\_(ツ)_/¯

Fine to rework to avoid though ...

Reply:
There shouldn't be enough fields to make the perf difference matter. Whichever logic is easier to follow would be my preference and I'll leave that to you 🙂

Reply:
> Btw. when `this.props.children` is not an array, now we return the full `this.state.data` object. Is that expected?

Returning the full `this.state.data` object makes this function a no-op (we're back to the original behavior prior to this exercise). I'm not super-familiar with all of the types of things React might put in `props.children`. Seems like if children is not an Array then falling back to the original behavior is as good an option as any. Eh?

Reply:
>Seems like if children is not an Array then falling back to the original behavior is as good an option as any. Eh?

Yup, sounds good to me. Thanks for explaining! 🎉